src/api/process.c

```diff
@@ -38,9 +44,24 @@ typedef struct {
     bool reading[2];
     char buffer[2][READ_BUF_SIZE];
   #endif
-  process_handle child_pipes[3][2];
+  process_stream_handle child_pipes[3][2];
 } process_t;
 
+typedef struct process_kill_s {
+  int tries;
+  uint32_t start_time;
+  process_handle handle;
+  struct process_kill_s *next;
+} process_kill_t;
+
+typedef struct {
+  bool stop;
+  SDL_mutex *mutex;
+  SDL_cond *has_work, *work_done;
```

Comment:
What is `work_done` used for?

Reply:
It is used to signal work done.
If a process is abandoned or killed, it will be signaled. The waiter waits until the list is empty.

Reply:
I don't think it's ever signaled tho.